### ISSUE TITLE
PushNotification does not appear to add device token to endpoint

### ISSUE THREAD

**Issue by dantasfiles** [aws-amplify/amplify-js#4441]
I've added Push Notifications to an empty React Native test project.
I'm experimenting on Android at the moment.
```
    "@aws-amplify/pushnotification": "^1.1.4",
    "amazon-cognito-identity-js": "^3.2.0",
    "aws-amplify": "^1.2.4",
    "aws-amplify-react-native": "^2.2.3",
    "react": "16.9.0",
    "react-native": "0.61.4"
```

My App.js looks like this:
```
import React, {useEffect} from 'react';
import Amplify from 'aws-amplify';
import awsconfig from './aws-exports';
import PushNotification from '@aws-amplify/pushnotification';
Amplify.Logger.LOG_LEVEL = 'DEBUG';
Amplify.configure(awsconfig);
PushNotification.configure(awsconfig);
...
function App(props) {
  useEffect(() => {
    PushNotification.onNotification(notification => {
      console.log('in app notification', notification);
    });
    PushNotification.onRegister(token => {
      console.log('in app registration', token);
    });
    PushNotification.onNotificationOpened(notification => {
      console.log('the notification is opened', notification);
    });
  }, []);
  ...
}
export default App;
```

When I look at the logfiles, the only line that appears interesting is:
```
 LOG  [DEBUG] 03:54.327 AWSPinpointProvider - updateEndpoint with params:  {"ApplicationId": "xxxxxxxx", "EndpointId": "xxxxxxxxxx", "EndpointRequest": {"Attributes": {}, "ChannelType": undefined, "Demographic": {"AppVersion": "/", "Make": "ReactNative", "Model": "", "ModelVersion": "", "Platform": undefined}, "EffectiveDate": "2019-11-21T21:03:54.325Z", "Location": {}, "Metrics": {}, "RequestId": "xxxxxxxx", "User": {"UserAttributes": [Object], "UserId": "..."}}}
```
There are no "Notification" debug lines.

From looking at [PushNotification.ts](https://github.com/aws-amplify/amplify-js/blob/master/packages/pushnotification/src/PushNotification.ts) I believe that the PushNotification.configure function should later update that same endpoint to add the Token and ChannelType, but it doesn't appear to do that.

When I run a test:
```
aws pinpoint get-endpoint --application-id xxxxxx --endpoint-id xxxxxx
```
the result is
```
{
    "EndpointResponse": {
        "ApplicationId": "xxxxxxxxx",
        "Attributes": {},
        "CohortId": "21",
        "CreationDate": "2019-11-21T21:03:54.325Z",
        "Demographic": {
            "AppVersion": "/",
            "Make": "ReactNative",
            "Model": "",
            "ModelVersion": ""
        },
        "EffectiveDate": "2019-11-21T21:03:54.325Z",
        "EndpointStatus": "ACTIVE",
        "Id": "xxxxxxx",
        "Location": {},
        "Metrics": {},
        "OptOut": "ALL",
        "RequestId": "xxxxxxx",
        "User": {
            "UserId": "xxxxxxx"
        }
    }
}
```
 with no device token or channel set.

As a result, if I try to send a test push notification using the Pinpoint console to that endpoint, it errors "Endpoints must have an address to target."
However, the Firebase console can see my mobile device and send test push notifications to it.

Why is the endpoint not being updated properly in Pinpoint?

**Comment by idanlo** [comment]
Why did you close this? Did you find a solution?

**Comment by dantasfiles** [comment]
It was caused by this overall issue
https://github.com/aws-amplify/amplify-js/issues/4351#issuecomment-557711904
And fixed in the current version of the aws-amplify package.

**Comment by idanlo** [comment]
is PushNotification supposed to add the device token by itself to the pinpoint endpoint address? Currently in my app I'm doing it manually.

**Comment by dantasfiles** [comment]
@idanlo 
Yes, Amplify updates the endpoint with the device token automatically.
If you turn debug logging on
`Amplify.Logger.LOG_LEVEL = 'DEBUG';`
You can see it happening.

**Comment by idanlo** [comment]
Doesn't work for me. When I'm trying to run `Analytics.updateEndpoint` I get `BadRequestException: Missing Address`, only when setting the address manually using the `PushNotification.onRegister` event it works.

**Comment by idanlo** [comment]
`PushNotification.updateEndpoint` doesn't work as well, it has to be `Analytics.updateEndpoint` where the address is the notification token

**Comment by dantasfiles** [comment]
I don't understand what you wrote. You shouldn't have to run `Analytics.updateEndpoint` or `PushNotification.updateEndpoint.` If you look at the debug log files on Android, you should see a line about the device token being received from Google or something like that, and you should see another line that the endpoint was set successfully. You can then send test messages to your endpoint from the pinpoint console. You don't have to call anything other than `Amplify.configure `

I haven't tried on iOS yet.

**Comment by idanlo** [comment]
well maybe it works just on Android then. Because when I try to run `Analytics.updateEndpoint` with user data (userId, attributes, etc), I get an error saying that the endpoint is missing an address (which is the notification token)

**Comment by dantasfiles** [comment]
That's weird, it should behave similarly for iOS and Android.

I think this is the line that's supposed to automatically update the endpoint on ios
https://github.com/aws-amplify/amplify-js/blob/867412030de57fd74078b609252de6f7f81ad331/packages/pushnotification/src/PushNotification.ts#L162

If it's not working, you might want to file an issue.

**Comment by MR-Neto** [comment]
@idanlo, I am also trying to use `Analytics.updateEndpoint` or `PushNotification.updateEndpoint` to update the userId. But I need the device token for the address. Do you have any insights?

**Comment by idanlo** [comment]
@MR-Neto no, still stuck with this. Analytics doesn’t work

**Comment by ArionMiles** [comment]
`Amplify.configure(awsconfig);`

should be in your `index.js` file, not `App.js`

**Comment by my-name-is-nheo** [comment]
following the docs did me no good. i'm able to get token for android but not for ios. did any one come up with a solution?

**Comment by Munish-GitHub** [comment]
Hi guys, My endpoint was not getting updated with aws pinpoint. Then i added firebase code in my react-native app and got the token from firebase. I tried using Analytics.updateEndpoint and passing the device token and it set the Pinpoint and I can send notification using Test Messaging on pinpoint console. Its successful. It does not work with Endpoint though. I tried getting endpoint using below code in react-native:  

Analytics.getPluggable('AWSPinpoint')._config.endpointId;

And i tried setting it... it sets succssfully but while sending the message it gives error:

Message not sent: We weren't able to send your push message to XXXXXXXXXXXXXXXX.
Details

Request Id 99 : Endpoints must have an address to target.


Anyone knows about the solution with Endpoint.

Below is my code:

Amplify.Logger.LOG_LEVEL = 'DEBUG';

Amplify.configure({
  Auth: {
      // REQUIRED only for Federated Authentication - Amazon Cognito Identity Pool ID
      identityPoolId: 'ap-XXXXXXX',

      // REQUIRED - Amazon Cognito Region
      region: 'ap-XXXXXX',

      // OPTIONAL - Amazon Cognito Federated Identity Pool Region 
      // Required only if it's different from Amazon Cognito Region
      identityPoolRegion: 'ap-XXXXXX',

      // OPTIONAL - Amazon Cognito User Pool ID
      userPoolId: 'ap-XXXXXXXX',
      

      // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
      userPoolWebClientId: 'XXXXXXXXXXXX',
  },
  Analytics: {
    disabled: false,
    AWSPinpoint: {
        appId: 'pinpoint app id',
        region: 'XXXXX',
    }
  },
});

Analytics.configure({
  appId: 'pinpoint app id',
});

PushNotification.configure({
  appId: 'pinpoint app id',
  region: 'region',
});

PushNotification.onNotification((notification) => {
  console.log('in app notification', notification);
});

// get the registration token
// This will only be triggered when the token is generated or updated.
PushNotification.onRegister((token) => {
  setToken(token);
  console.log('in app registration, token stored ', token);
});

async function setToken(token){
  await AsyncStorage.setItem("deviceToken", token);
  console.log('token set in storage ', token);
}

// get the notification data when notification is opened
PushNotification.onNotificationOpened((notification) => {
    console.log('the notification is opened', notification);
});


When my user is signed in, I am doing following:

//Token I get from FCM.
let deviceToken = await AsyncStorage.getItem("fcmToken");

const signin = async () => {
await Analytics.updateEndpoint({
          address: deviceToken,
          channelType: 'GCM',
          userId: "1",
        });
}


If i use direct device token from FCM it works but same does not work with endpoint. Can anyone help
